### PR TITLE
Feat shop 1586 upgrade packages to node 16

### PR DESCRIPTION
**Pull request by vikas-mindpath**
SHOP-1586 https://withjoy.atlassian.net/browse/SHOP-1586

### CODE REVIEW

**Comment by vikas-mindpath** [comment]
node 6 broken I need to figure out but rest you can review.

File: package.json [hunk 1]
```diff
@@ -77,28 +77,28 @@
...
     "typeorm": "~0.2.13",
-    "typescript": "^3.7.3",
+    "typescript": "^4.6.3",
     "url-regexp": "^1.0.2",
     "uuid": "^8.3.2"
   },
   "devDependencies": {
     "@types/chai": "^4.2.12",
-    "@types/jest": "^25.2.3",
+    "@types/jest": "^29.2.0",
```
Comment (cantremember) [inline package.json:86]: can we make this "^26" please, to match server-core-test

hmm.  well, actually ... you can keep thes jest upgrades, assuming that

- the Test Suite passes (it does)
- as part of https://github.com/joylifeinc/server-core-test/pull/22 , upgrade server-core-test to depend upon this (not yet published) version of server-core, and ensure its Test Suite passes
- ensure that the `photo_service` (arbitrary choice) Test Suite passes with upgrades to both of those (unublished) packages

if that all works, then this version of jest is unlikely to cause broader problems

and i should be clear ... we still need to cover all 3 of those bases above anyway, even if you don't upgrade jest

File: tsconfig.build.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     // "noImplicitOverride": true,
     "noImplicitReturns": true,
     "forceConsistentCasingInFileNames": true,
-    // "useUnknownInCatchVariables": false,
+    "useUnknownInCatchVariables": false
```
Comment (cantremember) [inline tsconfig.build.json:30]: 👍

**Review by cantremember — APPROVED** [review]
## you're hitting the right points

however, the Test Suite fails

once that's fixed -- we need to make sure that these new modules you've upgrading both work together in tandem in one of our Services -- as mentioned below.  it's the Final Step

if that all works, this is ✅ and i don't need to re-👀 it

please publish this as a MINOR releaes

**Pull request merged**